multicluster/test/e2e/framework.go

```diff
@@ -101,106 +79,130 @@ func (data *TestData) createClients() error {
+		data.clusterTestDataMap[cluster] = testData
+	}
+	return nil
+}
+
+func (data *MCTestData) initProviders() error {
+	for cluster, d := range data.clusterTestDataMap {
+		if err := d.InitProvider("remote", "multicluster"); err != nil {
+			log.Errorf("Failed to initialize provider for cluster %s", cluster)
+			return err
 		}
-		data.kubeconfigs[cluster] = kubeConfig
-		data.clients[cluster] = clusterClient
-		data.crdClients[cluster] = crdClient
 	}
+	provider, _ = providers.NewRemoteProvider("multicluster")
 	return nil
 }
 
-func (data *TestData) createTestNamespace() error {
-	for _, client := range data.clients {
-		if err := createNamespace(client, multiClusterTestNamespace, nil); err != nil {
+func (data *MCTestData) createTestNamespaces() error {
+	for cluster, d := range data.clusterTestDataMap {
+		if err := d.CreateNamespace(multiClusterTestNamespace, nil); err != nil {
+			log.Errorf("Failed to create test namespace for cluster %s", cluster)
 			return err
 		}
 	}
 	return nil
 }
 
-func (data *TestData) deletePod(clusterName string, namespace string, name string) error {
-	client := data.getClientOfCluster(clusterName)
-	return deletePod(client, namespace, name)
+func (data *MCTestData) deletePod(clusterName string, namespace string, name string) error {
+	if d, ok := data.clusterTestDataMap[clusterName]; ok {
+		if err := d.DeletePod(namespace, name); err != nil {
+			return err
+		}
+	}
+	return nil
 }
 
-func (data *TestData) deleteService(clusterName string, namespace string, name string) error {
-	client := data.getClientOfCluster(clusterName)
-	return deleteService(client, namespace, name)
+func (data *MCTestData) deleteService(clusterName string, namespace string, name string) error {
+	if d, ok := data.clusterTestDataMap[clusterName]; ok {
+		if err := d.DeleteService(namespace, name); err != nil {
+			return err
+		}
+	}
+	return nil
 }
 
-func (data *TestData) deleteTestNamespace(timeout time.Duration) error {
-	return data.deleteNamespaceInAllClusters(multiClusterTestNamespace, timeout)
+func (data *MCTestData) getService(clusterName string, namespace string, name string) (*corev1.Service, error) {
+	if d, ok := data.clusterTestDataMap[clusterName]; ok {
+		return d.GetService(namespace, name)
+	}
+	return nil, fmt.Errorf("clusterName %s not found", clusterName)
 }
 
-func (data *TestData) deleteNamespace(clusterName string, namespace string, timeout time.Duration) error {
-	client := data.getClientOfCluster(clusterName)
-	return deleteNamespace(client, namespace, timeout)
+func (data *MCTestData) deleteTestNamespaces(timeout time.Duration) error {
+	var failedClusters []string
+	for cluster, d := range data.clusterTestDataMap {
+		if err := d.DeleteNamespace(multiClusterTestNamespace, timeout); err != nil {
+			failedClusters = append(failedClusters, cluster)
+		}
+	}
+	if len(failedClusters) > 0 {
+		return fmt.Errorf("failed to delete mcs test namespace in clusters %v", failedClusters)
```

Comment:
better to include the correct Namespace name:
`fmt.Errorf("failed to delete Namespace %s in clusters %v", multiClusterTestNamespace, failedClusters)`